### PR TITLE
Add beta support for custom error response policies for URL maps

### PR DESCRIPTION
**Pull request by daanheikens**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/18200

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added fields `customErrorResponsePolicy` and `defaultErrorResponsePolicy` to the `google_compute_url_map` resource
```

### CODE REVIEW

**Comment by daanheikens** [comment]
Hey @rileykarson, bumping up this one. 

One of my colleagues requested this one and it would really help us to enhance a solution we are designing.

**Comment by rileykarson** [comment]
Started generation, probably won't be done before I sign off for the day. Will try to get a full pass done Monday, sorry for the delay!

**Comment by daanheikens** [comment]
> Started generation, probably won't be done before I sign off for the day. Will try to get a full pass done Monday, sorry for the delay!

No worries! Thanks. I will process your feedback.

**Comment by nika-pr** [comment]
Does someone else have time to review this? We are waiting for this to get merged...

File: mmv1/products/compute/UrlMap.yaml
Comment (rileykarson) [inline mmv1/products/compute/UrlMap.yaml:318]: Should the subvalues here be required, or can they be omitted? If they can be omitted, can we do that in some of the tests?

File: mmv1/products/compute/UrlMap.yaml
Comment (rileykarson) [inline mmv1/products/compute/UrlMap.yaml:336]:
```suggestion
              name: 'errorResponseRule'
```

File: mmv1/products/compute/UrlMap.yaml
Comment (rileykarson) [inline mmv1/products/compute/UrlMap.yaml:489]:
```suggestion
                    name: 'errorResponseRule'
```

File: mmv1/products/compute/UrlMap.yaml
Comment (rileykarson) [inline mmv1/products/compute/UrlMap.yaml:2065]:
```suggestion
        name: 'errorResponseRule'
```

File: mmv1/products/compute/UrlMap.yaml
Comment (daanheikens) [reply mmv1/products/compute/UrlMap.yaml:318]: I believe it's an optional value yes. I will make it optional and include it in a test

**Comment by rileykarson** [comment]
(TestAccComputeInstanceNetworkIntefaceWithSecurityPolicy is probably unrelated and you can ignore it unless I flag it specifically)

**Review by rileykarson — APPROVED** [review]

**Pull request merged**
